src/Diagnostics.RuntimeHost/Program.cs

```diff
@@ -13,12 +16,26 @@ public static void Main(string[] args)
 
         public static IWebHost BuildWebHost(string[] args)
         {
-            var config = new ConfigurationBuilder()
-                .AddCommandLine(args)
-                .Build();
-
             return WebHost.CreateDefaultBuilder(args)
-                .UseConfiguration(config)
+                .ConfigureAppConfiguration((context, config) =>
+                {
+                    var builtConfig = config
+                        .AddCommandLine(args)
+                        .Build();
+
+                    if (context.HostingEnvironment.IsDevelopment())
+                    {
+                        var azureServiceTokenProvider = new AzureServiceTokenProvider();
+                        var keyVaultClient = new KeyVaultClient(
+                            new KeyVaultClient.AuthenticationCallback(
+                                azureServiceTokenProvider.KeyVaultTokenCallback));
+
+                        config.AddAzureKeyVault(
+                            $"https://appservicediagnostics.vault.azure.net/",
```

Comment:
Since this is copied around a few times, we should store it in a constants file in case it needs to change